docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/QueryIou.daml

```diff
@@ -0,0 +1,15 @@
+{-# LANGUAGE ApplicativeDo #-}
+
+module Tests.QueryIou where
+
+import Daml.Script
+
+import Iou
+
+queryIou : Party -> ContractId Iou -> Script (Optional Iou)
```

Comment:
I think a better solution for this is to make a template that you can call via `createAndExercise` that provides a choice that calls `Fetch`. That provides a much more direct solution for translating arbitrary `Update` expressions passed to `submit` in scenarios to DAML Script.

Reply:
Initially I went with the `Fetch` approach. I thought it was a bit odd to have to define an extra template just for `fetch` when the functionality already exists in `query`, but I'm happy to change it back.

Reply:
The reason why I prefer the `Fetch` approach is that it works for arbitrary `Update` expressions, e.g., if you look at some of the monadic `Update` statements that you converted you got lucky that they were exactly `createAndExercise` but if they are something else, defining a choice for this works for everything whether it be `fetch`, monadic updates or anything else.

There is also a difference in that `query` does the filtering on the client side while `fetch` does it on the ledger.